src/mc/mcsim.f03

```diff
@@ -270,11 +365,20 @@ pure function list_confinement()
                       wlc_AB(I) = wlc_ABP(I)
                  ENDdo
              endif
+             if (MCTYPE==13) then 
+                wlc_basepairs = wlc_basepairs_prop
+            endif
              if(MCTYPE /= 7) then
-                 do I = 1,wlc_nPointsMoved
-                     J = wlc_pointsMoved(I)
-                     call updateR(J)
-                 enddo
+                if (WLC_P__GJK_STERICS) then 
+                    do I = left, right
+                        call updateR(I,left,right)
+                    enddo
+                else
+                    do I = 1,wlc_nPointsMoved
+                        J = wlc_pointsMoved(I)
+                        call updateR(J,0,0)
+                    enddo
+                endif
```

Comment:
Once again I think we may be able to combine points moved and left-right.  Sorry about bad indenting.

Reply:
so the reason i had to differentiate between left-tight and moved points is because the sterics rely on the convex hulls formed from the beads from i to i+1. to make sure the new moves conserve the correct shapes, i need to in-order iteratively update the positions/orientation vectors. so that the new convex hull vertices of the moved beads are always up-to-date. so that's why i go from left to right. let me know if this remains unclear!